### ISSUE TITLE
[no-unused-vars] false negative in import

### ISSUE THREAD

**Issue by kushalmahajan** [typescript-eslint/typescript-eslint#941]
Here's my code 

```
import { Link } from 'react-router-dom';
...

return (
...
  <li>
      <Link to="/">Home</Link>
  </li>
) 
```
The import statement reports an error unnecesarily `'Link' is defined but never used.eslint(@typescript-eslint/no-unused-vars)`

Here are my settings

```
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "project": "./tsconfig.json"
    },
    "plugins": [
        "react",
        "prettier",
        "@typescript-eslint"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier",
        "prettier/@typescript-eslint",
        "plugin:@typescript-eslint/recommended-requiring-type-checking"
    ],
    "env": {
        "browser": true,
        "jasmine": true,
        "jest": true
    },
    "rules": {
        "prettier/prettier": [
            "error",
            {
                "singleQuote": true,
                "jsx-max-props-per-line": 1,
                "tabWidth": 4
            }
        ],
        "quotes": "off",
        "@typescript-eslint/quotes": [
            "error",
            "single",
            {
                "avoidEscape": true
            }
        ],
        "no-unused-vars": "off",
        "@typescript-eslint/no-unused-vars": [
            "error",
            {
                "vars": "all",
                "args": "all",
                "ignoreRestSiblings": false
            }
        ],
        "@typescript-eslint/no-explicit-any": [
            "error",
            {
                "fixToUnknown": true,
                "ignoreRestArgs": false
            }
        ],
        "@typescript-eslint/explicit-function-return-type": [
            "error",
            {
                "allowExpressions": true
            }
        ]
    },
    "settings": {
        "react": {
            "pragma": "React",
            "version": "detect"
        }
    },
}
```
It's a bug clearly.

**Comment by bradzacher** [comment]
You should use `eslint-plugin-react`, it has a rule to mark usages of JSX.

**Comment by snokier** [comment]
I use `eslint-plugin-react`, but got this error too. My `.eslintrc` is:

```
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["react", "@typescript-eslint"],
  "extends": ["plugin:@typescript-eslint/recommended"],
  "parserOptions": {
    "ecmaVersion": 2019,
    "sourceType": "module",
    "ecmaFeatures":{
      "jsx": true
    }
  },
  "rules": {
    "@typescript-eslint/explicit-function-return-type": 0,
    "@typescript-eslint/no-explicit-any": 0,
  }
}
```

**Comment by snokier** [comment]
I got the same error with `eslint-plugin-react`. My `.eslintrc`:

```
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["react", "@typescript-eslint"],
  "extends": ["plugin:@typescript-eslint/recommended"],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  }
}
```

When I add `"plugin:react/recommended"` to `extends`(`"extends": ["plugin:@typescript-eslint/recommended", "plugin:react/recommended"]`), this error is gone. But I get other errors, such as `'className' is missing in props validation`.

```
interface ShortcutProps {
  className?: string;
}
const Shortcut: React.FC<ShortcutProps> = ({ className }) => {}
```

**Comment by betopompolo** [comment]
@snokier I don't know if you already solved the issue you described, but I believe if you type your props, the error is gone. Something like this
```ts
interface ShortcutProps {
  className?: string;
}
const Shortcut: React.FC<ShortcutProps> = (props: ShortcutProps) => {
   // code...
}
```
